httpclient/src/main/java/feign/httpclient/ApacheHttpClient.java

```diff
@@ -153,13 +153,17 @@ HttpUriRequest toHttpUriRequest(Request request, Request.Options options) throws
   private ContentType getContentType(Request request) {
     ContentType contentType = ContentType.DEFAULT_TEXT;
     for (Map.Entry<String, Collection<String>> entry : request.headers().entrySet())
-    if (entry.getKey().equalsIgnoreCase("Content-Type")) {
-      Collection values = entry.getValue();
-      if (values != null && !values.isEmpty()) {
-        contentType = ContentType.create(entry.getValue().iterator().next(), request.charset());
-        break;
+      if (entry.getKey().equalsIgnoreCase("Content-Type")) {
+        Collection<String> values = entry.getValue();
+        if (values != null && !values.isEmpty()) {
+          String mimeType = values.iterator().next();
+          contentType = ContentType.parse(mimeType);
+          if (contentType.getCharset() == null) {
+            contentType = ContentType.create(contentType.getMimeType(), request.charset());
```

Comment:
this was a bad recommendation of me. we shouldn't conflate the response charset with the request one.

Please simplify the code to the following vs pretending we know what the response charset is.. You'll notice all tests pass!

```
      if (values != null && !values.isEmpty()) {
        contentType = ContentType.parse(entry.getValue().iterator().next());
        break;
      }
```